src/language/code_generator.cmake

```diff
@@ -0,0 +1,222 @@
+#
+# THIS FILE IS GENERATED AND SHALL NOT BE EDITED.
```

Comment:
probably a stupid question - but if this is generated, does it need to be committed to the repo?

Reply:
Allow me copy/paste my PR description:

> Code generator creates a new file: src/language/code_generator.cmake
It defines a set CMake variables to be used elsewhere in the CMake
project to manipulate inputs and outputs of the code generator.
This file is updated by the code generator if a Jinja template is added
or removed, or is a NMODL AST class is added for instance.
The file is added to revision control on purpose to break the cyclic dependency
but is not meant to be edited manually.

Description of the cyclic dependency.
1. This file is required by CMake
1. This file is generated by code_generator.py
1. code_generator.py is driven by CMake

Add the file to the repo is not ideal but that's fine IMO